docs/resources/kustomization.md

```diff
@@ -0,0 +1,184 @@
+Only one scope per resource is allowed.
+
+## Cluster group scoped Kustomization
+
+### Example Usage
+
+```terraform
+# Create Tanzu Mission Control kustomization with attached set as default value.
+resource "tanzu-mission-control_kustomization" "create_cluster_group_kustomization" {
+  name = "tf-kustomizationy-name" # Required
+
+  namespace_name = "tf-namespace" #Required
+
+  scope {
+    cluster_group {
+      name = "default" # Required
+    }
+  }
+
+  meta {
+    description = "Create namespace through terraform"
+    labels      = { "key" : "value" }
+  }
+
+  spec {
+    path = "testPath" # Required
+    prune = "testPrune"
+    interval = "10m" # Default: 5m
+    target_namespace = "testTargetNamespace"
+    source {
+        name = "testGitRepositoryName" # Required
+        namespace = "testGitRepositoryNamespace" # Required
+    }
+  }
+}
+```
+
+## Cluster scoped Kustomization
+
+### Example Usage
+
+```terraform
+# Create Tanzu Mission Control kustomization with attached set as default value.
+resource "tanzu-mission-control_kustomization" "create_cluster_kustomization" {
+  name = "tf-kustomizationy-name" # Required
+
+  namespace_name = "tf-namespace" #Required
+
+  scope {
+    cluster {
+      cluster_name            = "testcluster" # Required
+      provisioner_name        = "attached"    # Default: attached
+      management_cluster_name = "attached"    # Default: attached
+    }
+  }
+
+  meta {
+    description = "Create namespace through terraform"
+    labels      = { "key" : "value" }
+  }
+
+  spec {
+    path = "testPath" # Required
+    prune = "testPrune"
+    interval = "10m" # Default: 5m
+    target_namespace = "testTargetNamespace"
+    source {
+        name = "testGitRepositoryName" # Required
+        namespace = "testGitRepositoryNamespace" # Required
+    }
+  }
+}
+```
+
+<!-- schema generated by tfplugindocs -->
+## Schema
+
+### Required
+
+- `name` (String) Name of the Kustomization.
+- `namespace_name` (String) Name of Namespace.
+- `scope` (Block List, Min: 1, Max: 1) Scope for the kustomization, having one of the valid scopes: cluster, cluster_group. (see [below for nested schema](#nestedblock--scope))
+- `spec` (Block List, Min: 1, Max: 1) Spec for the Repository. (see [below for nested schema](#nestedblock--spec))
+
+### Optional
+
+- `meta` (Block List, Max: 1) Metadata for the resource (see [below for nested schema](#nestedblock--meta))
+
+### Read-Only
+
+- `id` (String) The ID of this resource.
+- `status` (Map of String) Status for the Kustomization.
+
+<a id="nestedblock--scope"></a>
+### Nested Schema for `scope`
+
+Optional:
+
+- `cluster` (Block List, Max: 1) The schema for cluster iam policy full name (see [below for nested schema](#nestedblock--scope--cluster))
+- `cluster_group` (Block List, Max: 1) The schema for cluster group iam policy full name (see [below for nested schema](#nestedblock--scope--cluster_group))
```

Comment:
Why is kustomization documentation having IAM policy?

Reply:
removed, updated the docs